packages/cli/e2e/__tests__/trigger.spec.ts

```diff
@@ -65,4 +65,20 @@ describe('trigger', () => {
     expect(result.stdout).toContain('No matching checks were found.')
     expect(result.status).toBe(0)
   })
+
+  test('Should return code 1 when no checks match and the fail-on-no-match flag is set', async () => {
+    const result = await runChecklyCli({
+      args: [
+        'trigger',
+        '--tags',
```

Comment:
It may be trying to treat the `--fail-on-no-matching` flag as a tag right now because `--tags` is not given a value.

```suggestion
```

Reply:
Ah yah, good call. I have fixed the test I think!